source/include/TypeAndDataManager.h

```diff
@@ -1245,7 +1242,7 @@ class String : public TypedArray1D< Utf8Char >
     void AppendEscapeEncoded(const Utf8Char* source, IntIndex len);
 
     void InsertSubString(IntIndex position, SubString* str) {
-        Insert(position, (IntIndex)str->Length(), (Utf8Char*)str->Begin());
+        Insert(position, static_cast<IntIndex>(str->Length()), const_cast<Utf8Char*>(str->Begin()));
```

Comment:
I think const_cast<Utf8Char*>is not required here. Just remove const_cast and even C style cast and try "str->Begin()"?